pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -1400,6 +1406,12 @@ public void openLedgerComplete(ManagedLedger ledger, Object ctx) {
                                             "Replication or dedup check failed."
                                                     + " Removing topic from topics list {}, {}",
                                             topic, ex);
+                                    persistentTopic.getTransactionBuffer()
```

Comment:
Should we add this similar logic to line `1395`?

Reply:
Great catch!
it makes sense, in line `1395`, it seems like checking create topic timeout, it's also need to close transaction buffer.
I'll create another PR to add the logic